client/app/components/OnboardingTopBar/OnboardingTopBar.js

```diff
@@ -4,22 +4,51 @@ import { div, span, a } from 'r-dom';
 
 import css from './OnboardingTopBar.scss';
 
-const next = function next(nextStep, guideRoot, t) {
-  return t(nextStep) ? {
-    title: t(nextStep),
-    link: `${guideRoot}/${nextStep}`,
-  } : null;
+const next = function next(nextStep, guideRoot) {
+  switch (nextStep) {
+  case "slogan_and_description":
+    return {
+      title: I18n.t("admin.onboarding.topbar.slogan_and_description"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "cover_photo":
+    return {
+      title: I18n.t("admin.onboarding.topbar.cover_photo"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "filter":
+    return {
+      title: I18n.t("admin.onboarding.topbar.filter"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "paypal":
+    return {
+      title: I18n.t("admin.onboarding.topbar.paypal"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "listing":
+    return {
+      title: I18n.t("admin.onboarding.topbar.listing"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  case "invitation":
+    return {
+      title: I18n.t("admin.onboarding.topbar.invitation"),
+      link: `${guideRoot}/${nextStep}`
+    };
+  default:
+    return null;
+  }
```

Comment:
For greppability. The current RoR has tons of old unused translations in the translation files, because the `t` function is called with a variable. So I'd prefer hard-coding the translations keys so that they can be grep'd and removed in the future.